crates/sui-indexer/migrations/pg/2023-10-06-204335_tx_indices/up.sql

```diff
@@ -1,54 +1,74 @@
+    PRIMARY KEY(object_id, tx_sequence_number)
 );
+CREATE INDEX tx_input_objects_sender ON tx_input_objects (sender, object_id, tx_sequence_number);
 
 CREATE TABLE tx_changed_objects (
-    cp_sequence_number          BIGINT       NOT NULL,
     tx_sequence_number          BIGINT       NOT NULL,
-    -- Object Id in bytes.
     object_id                   BYTEA        NOT NULL,
-    PRIMARY KEY(object_id, tx_sequence_number, cp_sequence_number)
+    sender                      BYTEA        NOT NULL,
+    PRIMARY KEY(object_id, tx_sequence_number)
+);
+CREATE INDEX tx_changed_objects_sender ON tx_changed_objects (sender, object_id, tx_sequence_number);
+
+CREATE TABLE tx_calls_pkg (
+    tx_sequence_number          BIGINT       NOT NULL,
+    package                     BYTEA        NOT NULL,
+    sender                      BYTEA        NOT NULL,
+    PRIMARY KEY(package, tx_sequence_number)
 );
+CREATE INDEX tx_calls_pkg_sender ON tx_calls_pkg (sender, package, tx_sequence_number);
 
-CREATE TABLE tx_calls (
-    cp_sequence_number          BIGINT       NOT NULL,
+CREATE TABLE tx_calls_mod (
+    tx_sequence_number          BIGINT       NOT NULL,
+    package                     BYTEA        NOT NULL,
+    module                      TEXT         NOT NULL,
+    sender                      BYTEA        NOT NULL,
+    PRIMARY KEY(package, module, tx_sequence_number)
+);
+CREATE INDEX tx_calls_mod_sender ON tx_calls_mod (sender, package, module, tx_sequence_number);
+
+CREATE TABLE tx_calls_fun (
     tx_sequence_number          BIGINT       NOT NULL,
     package                     BYTEA        NOT NULL,
     module                      TEXT         NOT NULL,
     func                        TEXT         NOT NULL,
-    -- 1. Using Primary Key as a unique index.
-    -- 2. Diesel does not like tables with no primary key.
-    PRIMARY KEY(package, tx_sequence_number, cp_sequence_number)
+    sender                      BYTEA        NOT NULL,
+    PRIMARY KEY(package, module, func, tx_sequence_number)
 );
-CREATE INDEX tx_calls_module ON tx_calls (package, module, tx_sequence_number, cp_sequence_number);
-CREATE INDEX tx_calls_func ON tx_calls (package, module, func, tx_sequence_number, cp_sequence_number);
-CREATE INDEX tx_calls_tx_sequence_number ON tx_calls (tx_sequence_number, cp_sequence_number);
+CREATE INDEX tx_calls_fun_sender ON tx_calls_fun (sender, package, module, func, tx_sequence_number);
 
--- un-partitioned table for tx_digest -> (cp_sequence_number, tx_sequence_number) lookup.
 CREATE TABLE tx_digests (
     tx_digest                   BYTEA        PRIMARY KEY,
-    cp_sequence_number          BIGINT       NOT NULL,
     tx_sequence_number          BIGINT       NOT NULL
 );
+
+CREATE TABLE tx_kinds (
+    tx_sequence_number          BIGINT       NOT NULL,
+    tx_kind                     SMALLINT     NOT NULL,
+    PRIMARY KEY(tx_kind, tx_sequence_number)
+);
+
+CREATE TABLE cp_tx (
```

Comment:
It's probably easier to add these as columns on the `checkpoints` table -- I introduced the new table for backfill just to avoid having to copy out the checkpoint data.